From 014b24e96f87d39812b039682d924647b58bd25f Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Thu, 16 Jul 2015 11:18:15 +0530
Subject: [PATCH] USBC_PHY: Fixed mutex deadlock issue during reset in
 phy-fusb300

The fusb300_reset_pd function is called from fusb300_phy_reset with
mutex lock held which is causing deadlock. This patch will relase
the lock before calling fusb300_reset_pd.

Change-Id: Ie353972b998287c42d674b2781be4ea4a53a7957
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13620
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/usb/typec/phy-fusb300.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index b5de90179e5b..0bc639651cb8 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -987,8 +987,8 @@ static int fusb300_phy_reset(struct typec_phy *phy)
 		fusb300_pd_send_hard_rst(phy);
 	else
 		fusb302_pd_send_hard_rst(phy);
-	fusb300_reset_pd(phy);
 	mutex_unlock(&chip->lock);
+	fusb300_reset_pd(phy);
 	return 0;
 }
 
-- 
1.9.1

